src/Core/Resolvers/QueryExecutor.cs

```diff
@@ -65,12 +64,23 @@ public class QueryExecutor<TConnection> : IQueryExecutor
             IDictionary<string, DbConnectionParam> parameters,
             Func<DbDataReader, List<string>?, Task<TResult>>? dataReaderHandler,
             HttpContext? httpContext = null,
-            List<string>? args = null)
+            List<string>? args = null,
+            string dataSourceName = "")
         {
+            if (string.IsNullOrEmpty(dataSourceName))
+            {
+                dataSourceName = ConfigProvider.GetConfig().DefaultDataSourceName;
+            }
+
+            if (!ConnectionStringBuilders.ContainsKey(dataSourceName))
```

Comment:
Since this parameter is passed in through a function call it could definitely be invalid. In our logic, we would retrieve the entity name from the entityNametoDbName dict and then pass it in to this function. However, if there is some logic error and we pass in the wrong name (for example, we send in some other variable due to a code error instead of one retrieved from entityNametodbName), this should throw exception. Even test methods in future that may want to test against this method could always by mistake provide wrong input.